packages/apps/reputation-oracle/server/src/modules/user/user.service.spec.ts

```diff
@@ -138,6 +154,190 @@ describe('UserService', () => {
     });
   });
 
+  describe('registerLabeler', () => {
+    it('should register labeler successfully and return site key', async () => {
+      const userEntity: DeepPartial<UserEntity> = {
+        id: 1,
+        email: MOCK_EMAIL,
+        evmAddress: MOCK_ADDRESS,
+        type: UserType.WORKER,
+        kyc: {
+          status: KycStatus.APPROVED,
+        },
+        save: jest.fn(),
+      };
+      const data = {
+        address: MOCK_ADDRESS,
+        chainId: ChainId.LOCALHOST,
+        type: OracleType.HCAPTCHA,
+        country: 'US',
+      };
+
+      const mockLabelerData = { sitekeys: [{ sitekey: 'site_key' }] };
+
+      (registerLabeler as jest.Mock).mockResolvedValueOnce(true);
+      (getLabelerData as jest.Mock).mockResolvedValueOnce(mockLabelerData);
+
+      web3Service.getSigner = jest.fn().mockReturnValue({
+        signMessage: jest.fn().mockResolvedValue('signature'),
+      });
+
+      const result = await userService.registerLabeler(
+        userEntity as UserEntity,
+        data,
+      );
+
+      expect(result).toEqual('signature');
+    });
+
+    it('should throw BadRequestException if user address is incorrect', async () => {
+      const userEntity: DeepPartial<UserEntity> = {
+        id: 1,
+        email: MOCK_EMAIL,
+        evmAddress: MOCK_ADDRESS,
+        type: UserType.WORKER,
+        kyc: {
+          status: KycStatus.APPROVED,
+        },
+        save: jest.fn(),
+      };
+
+      const data = {
+        address: 'different_address',
+        chainId: ChainId.LOCALHOST,
+        type: OracleType.HCAPTCHA,
+        country: 'US',
+      };
+
+      await expect(
+        userService.registerLabeler(userEntity as UserEntity, data),
+      ).rejects.toThrow(BadRequestException);
```

Comment:
Check the error message Rather than just validating the type of error thrown, we should also confirm that the error messages themselves are appropriate and useful. Same for the rest of the file

Reply:
Correct, thanks!